lib/rubocop/cop/sorbet/rbi_versioning/valid_gem_version_annotations.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Sorbet
+      # Checks that gem versions in RBI annotations are properly formatted per the Bundler gem specification.
+      #
+      # @example
+      #   # bad
+      #   # @version > not a version number
+      #
+      #   # good
+      #   # @version = 1
+      #
+      #   # good
+      #   # @version > 1.2.3
+      #
+      #   # good
+      #   # @version <= 4.3-preview
+      #
+      class ValidGemVersionAnnotations < Base
+        include GemVersionAnnotationHelper
+
+        MSG = "Invalid gem version(s) detected: %<versions>s"
+        VALID_OPERATORS = ["=", "!=", ">", ">=", "<", "<=", "~>"]
+
+        def on_new_investigation
+          gem_version_annotations.each do |comment|
+            invalid_versions = gem_versions(comment).select do |version|
+              !valid_version?(version)
+            end
+
+            unless invalid_versions.empty?
+              message = format(MSG, versions: invalid_versions.join(", "))
+              add_offense(comment, message: message)
+            end
+          end
+        end
+
+        private
+
+        def valid_version?(version_string)
+          parts = version_string.strip.split(" ")
+          operator, version = parts
+
+          return false if operator.nil? || parts.nil?
+
+          return false unless VALID_OPERATORS.include?(operator)
+
+          begin
+            Gem::Version.new(version)
```

Comment:
Wdyt of using the underlying method instead? https://github.com/rubygems/rubygems/blob/40784bb244f7471fa938a421e2773ec2c7a45405/lib/rubygems/version.rb#L173C12-L177.

It won't rely on exception throwing but it might be more brittle to future changes. Although I doubt the interface will go through frequent changes.

Reply:
Oh nice, that's a lot simpler! Thank you!